### PR TITLE
Use token when available

### PR DESCRIPTION
**Pull request by BrianCraig**
Hi!

I'm currently using openshift-rest-client, and i needed a way to log in with username & token, so i adapted the code to accept a token when present.

What do you think about this change?

### CODE REVIEW

**Comment by lholmquist** [comment]
@BrianCraig i will take a look at this in the next couple days

**Comment by lholmquist** [comment]
could you add a test for this?

**Comment by BrianCraig** [comment]
> could you add a test for this?

Hi! Maybe on the weekend i could do this, i can't ensure i will add one this weekend. We are using this change on a project currently, and its working as intended, but tests are needed to keep it up with the code quality. Greetings, Brian

File: lib/openshift-rest-client.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ async function openshiftClient (settings = {}) {
         const user = config.auth.username || config.auth.user;
         const password = config.auth.password || config.auth.pass;
 
-        const accessToken = await getTokenFromBasicAuth({ insecureSkipTlsVerify, url, user, password, authUrl });
+        const accessToken = config.auth.token ? config.auth.token : await getTokenFromBasicAuth({ insecureSkipTlsVerify, url, user, password, authUrl });
```
Comment (lholmquist) [inline lib/openshift-rest-client.js]: the only way to get to here is if you are also passing in a 'user' or 'username' in the `config.auth`,  so if your config object looked like this:
```
{
  auth: {
    token: 'asawefwfwf'
  }
}
```
then this code would't get run.  probably need to add a check a few lines up that a token is being passed

**Comment by lholmquist** [comment]
@BrianCraig are you still available to work on this?

**Comment by lholmquist** [comment]
Going to close this since it hasn't been worked on for a while.  if it needs to be re-opened we can do that

**Pull request closed without merging**